### PR TITLE
chore: add GitHub Actions VS Code extension

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Add the official GitHub Actions VS Code extension to devcontainer
- Add the official GitHub Actions VS code extension to the VS Code recommended extensions config

## Context

Works:

- Integrated docs

Broken:

- Syntax highlighting (at least in Codespaces)

## Important stuff

I'll let the maintainers decide if this is something they want. I'm marking this PR as "draft" so we don't accidentally merge it without proper review. 😄 

- The extension wants you to login to GitHub, thereby giving it more rights than we're probably comfortable with
- The extension is in beta
- The extension is from GitHub themselves

## References

- [GitHub Blog, GitHub Actions: Visual Studio Code Extension is now in public beta](https://github.blog/changelog/2023-03-28-github-actions-visual-studio-code-extension-is-now-in-public-beta/)
- [VSCode Marketplace, GitHub Actions](https://marketplace.visualstudio.com/items?itemName=github.vscode-github-actions)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "dbaeumer.vscode-eslint",
     "esbenp.prettier-vscode",
     "orta.vscode-jest",
-    "EditorConfig.editorconfig"
+    "editorconfig.editorconfig",
```
Comment (viceice) [inline .vscode/extensions.json:6]: why different casing?

Comment (HonkingGoose) [reply .vscode/extensions.json:6]: The CamelCase comes from the VS Code Marketplace unique identifier.

I think the lowercase comes from the VS Code option to put installed extensions in the configuration files.

I don't know if the casing is important. We don't want to get a different plugin due to different casing. 🙈

Comment (viceice) [reply .vscode/extensions.json:6]: in the other file it was uppercase. i think it's case insensitive. so we should lowercase all to be consistent.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -14,7 +14,8 @@
     "dbaeumer.vscode-eslint",
     "esbenp.prettier-vscode",
     "orta.vscode-jest",
-    "EditorConfig.editorconfig"
+    "EditorConfig.editorconfig",
+    "GitHub.vscode-github-actions"
```
Comment (HonkingGoose) [inline .devcontainer/devcontainer.json:18]:
```suggestion
    "editorconfig.editorconfig",
    "github.vscode-github-actions"
```

**Review by viceice — DISMISSED** [review]
🤷‍♂️

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -19,7 +19,8 @@
         "dbaeumer.vscode-eslint",
         "esbenp.prettier-vscode",
         "orta.vscode-jest",
-        "EditorConfig.editorconfig"
+        "EditorConfig.editorconfig",
```
Comment (secustor) [inline .devcontainer/devcontainer.json:22]:
```suggestion
        "editorConfig.editorconfig",
```
Should be lowercase based on the discussions in the other thread

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -19,7 +19,8 @@
         "dbaeumer.vscode-eslint",
         "esbenp.prettier-vscode",
         "orta.vscode-jest",
-        "EditorConfig.editorconfig"
+        "editorConfig.editorconfig",
```
Comment (HonkingGoose) [inline .devcontainer/devcontainer.json:22]:
```suggestion
        "editorconfig.editorconfig",
```

**Review by JamieMagee — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**